src/MusicTheory/PitchClass/Enharmonic.elm

```diff
@@ -0,0 +1,245 @@
+module MusicTheory.PitchClass.Enharmonic exposing
+    ( atMostOneAccidental
+    , exactSpelling
+    , naturalOrSingleFlat
+    , naturalOrSingleSharp
+    , simplified
+    , simplifiedToString
+    , toString
+    )
+
+import MusicTheory.Accidental exposing (Accidental(..))
+import MusicTheory.Internals.PitchClass as Internal
+import MusicTheory.Letter exposing (Letter(..))
+import MusicTheory.PitchClass as PitchClass exposing (PitchClass)
+
+
+{-| Returns the letter and accidental of a pitch class if the letter is raised or lowered such that it can be expressed in terms of a valid accidental.
+
+    (PitchClass.pitchClass F Sharp |> exactSpelling) == Just ( F, Sharp )
+
+-}
+exactSpelling : PitchClass -> Maybe ( Letter, Accidental )
+exactSpelling (Internal.PitchClass letter offset) =
+    offsetToAccidental offset
+        |> Maybe.map (Tuple.pair letter)
+
+
+atMostOneAccidental : PitchClass -> Maybe ( Letter, Accidental )
```

Comment:
It looks like I give this function a `PitchClass` that I want it to spell with at most one accidental, and it returns `Nothing` if it isn't spelled that way already. This doesn't seem useful to me.

I would expect that I give it a `PitchClass`, which may or may not be spelled the way I want, and it gives me the normalized `PitchClass` back, like this:

```
atMostOneAccidental (pitchClass A doubleSharp)
-> PitchClass B (Offset 0)
```
or:
```
atMostOneAccidental (pitchClass B natural)
-> PitchClass B (Offset 0)
```

Reply:
But A## is a valid pitch class. This is useful IMO. Of you want to do what you describe you have to use the function `simplified `.